compiler/AST/stmt.cpp

```diff
@@ -928,7 +928,16 @@ CondStmt::codegen() {
   codegenStmt(this);
 
   if ( outfile ) {
-    info->cStatements.push_back("if (" + codegenValue(condExpr).c + ") ");
+    //here it's very possible that we end up with ( ) around condExpr. Extra
+    //parentheses generated warnings from the backend compiler in some cases.
+    //It didn't feel very safe to strip them at expr level as it might mess up
+    //precedence thus, following conditional -- Engin
+
+    std::string c_condExpr = codegenValue(condExpr).c;
+    if (c_condExpr[0] == '(' && c_condExpr[c_condExpr.size()-1] == ')') {
+      c_condExpr = c_condExpr.substr(1, c_condExpr.length()-2);
+    }
```

Comment:
I'd make this a while loop.. why stop at removing 1 set of unnecessary parenthesis?